packages/toaster/components/Toaster.tsx

```diff
@@ -15,124 +15,96 @@ interface ToasterProps {
+    setToasts(toasts.filter(toast => id !== toast.props.id));
+  };
+
+  const restartTimeouts = () => {
+    toasts.forEach((toast: Toast) => {
+      const toastKey = toast.key as number;
+      if (toast.props.autodismiss) {
+        timeouts.current.push(
+          window.setTimeout(() => {
+            handleDismissToast(toast.props.id);
+          }, DELAY_TIME + MARGINAL_DELAY * toastKey)
+        );
+      }
+    });
+  };
 
-  public componentWillUnmount() {
-    this.clearTimeouts();
-  }
+  const clearTimeouts = () => {
+    timeouts.current.forEach(clearTimeout);
+  };
 
-  public componentDidMount() {
-    this.restartTimeouts();
-  }
+  React.useEffect(() => {
+    return () => {
+      clearTimeouts();
+    };
+  }, []);
 
-  public UNSAFE_componentWillReceiveProps(props: ToasterProps) {
-    const children = React.Children.toArray(props.children);
-    const childIds = children.map(toast => toast.props.id);
-    const currentIds = this.state.toasts.map(toast => toast.props.id);
+  React.useEffect(() => {
+    const childIds = toastChildren.map(toast => toast.props.id);
+    const currentIds = toasts.map(toast => toast.props.id);
 
     if (
       !currentIds.every(e => childIds.includes(e)) ||
       !currentIds.length ||
       childIds.length !== currentIds.length
     ) {
-      this.setState(
-        () => ({ toasts: children.map(this.cloneToast) }),
-        () => {
-          this.state.toasts
-            ?.filter(t => t.props.autodismiss)
-            .map(this.setTimer);
-        }
-      );
+      setToasts(toastChildren);
     }
-  }
-
-  public render() {
-    return (
-      <div
-        className={cx(toaster, margin("all"), marginAt.medium("all", "l"))}
-        data-cy="toaster"
+  }, [children]);
+
+  React.useEffect(() => {
+    restartTimeouts();
+  }, [toasts]);
+
+  return (
+    <div
+      className={cx(toaster, margin("all"), marginAt.medium("all", "l"))}
+      data-cy="toaster"
+    >
+      <ol
+        onMouseEnter={clearTimeouts}
+        onMouseLeave={restartTimeouts}
+        aria-live="assertive"
+        className={listReset}
+        data-cy="toaster-list"
       >
-        <ol
-          onMouseEnter={this.clearTimeouts}
-          onMouseLeave={this.restartTimeouts}
-          aria-live="assertive"
-          className={listReset}
-        >
-          <TransitionGroup>
-            {this.state.toasts.map(toast => (
-              <Transition
-                key={`toastWrapper-${toast.props.id}`}
-                timeout={{ enter: 0, exit: animationDuration }}
-              >
-                {state => (
-                  <li
-                    className={cx(
-                      preTransitionStyle(animationDuration),
-                      transitionStyles[state]
-                    )}
-                  >
-                    {toast}
-                  </li>
-                )}
-              </Transition>
-            ))}
-          </TransitionGroup>
-        </ol>
-      </div>
```

Comment:
component wrapped under React.memo should be given displayname to make debugging easier in react devtools. Else it will show up as Anonymous component